app/gui/src/project-view/components/ComponentActionButton.vue

```diff
@@ -0,0 +1,45 @@
+<script setup lang="ts">
+import MenuButton from '@/components/MenuButton.vue'
+import SvgIcon from '@/components/SvgIcon.vue'
+import { type ComponentAction } from '@/util/componentActions'
+
+const { action } = defineProps<{ action: ComponentAction }>()
+</script>
+
+<template>
+  <!-- eslint-disable vue/no-mutating-props -- `ComponentAction` is an internally-reactive object. -->
```

Comment:
AFAIK, even mutating internally-reactive object is an antipattern in Vue: such things should be done by models.

Here, I think we could just inject component actions, and pass only action key as a property.